resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/SubmittedDependencyUploaderImpl.scala

```diff
@@ -76,29 +82,32 @@ private[spark] class SubmittedDependencyUploaderImpl(
     Utils.tryWithResource(new FileOutputStream(filesTgz)) { filesOutputStream =>
       CompressionUtils.writeTarGzipToStream(filesOutputStream, files.map(_.getAbsolutePath))
     }
-    // TODO provide credentials properly when the staging server monitors the Kubernetes API.
-    val kubernetesCredentialsString = OBJECT_MAPPER.writer()
-      .writeValueAsString(KubernetesCredentials(None, None, None, None))
-    val labelsAsString = OBJECT_MAPPER.writer().writeValueAsString(podLabels)
+    val clientKeyBase64 = clientKeyFile.map(f => BASE_64.encode(Files.toByteArray(f)))
+    val clientCertBase64 = clientCertFile.map(f => BASE_64.encode(Files.toByteArray(f)))
+    val oauthTokenBase64 = oauthToken.map(token => BASE_64.encode(token.getBytes(Charsets.UTF_8)))
 
-    val filesRequestBody = RequestBody.create(
-      okhttp3.MediaType.parse(MediaType.MULTIPART_FORM_DATA), filesTgz)
-
-    val kubernetesCredentialsBody = RequestBody.create(
-      okhttp3.MediaType.parse(MediaType.APPLICATION_JSON), kubernetesCredentialsString)
+    val ownerMonitoringCredentials = StagedResourcesOwnerMonitoringCredentials(
+        clientKeyDataBase64 = clientKeyBase64,
+        clientCertDataBase64 = clientCertBase64,
+        oauthTokenBase64 = oauthTokenBase64)
+    val stagedResourcesOwner = StagedResourcesOwner(
+      ownerNamespace = podNamespace,
```

Comment:
There will also be issues when we use expiring credentials like oauth tokens; it seems like in general, storing cert data long-term in a single component is going to be dangerous IMO. I think we ought to encourage creating a role/cluster-role, and a service account allowing read access to the pods in the cluster that need to be tracked for this component. That's more idiomatic and works more broadly.